src/main/java/org/example/bookstoreapp/controller/BookController.java

```diff
@@ -28,43 +32,50 @@ public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Find all books", description = "Get a list of all available books")
+    public List<BookDto> getAll(Pageable pageable) {
```

Comment:
You return a List<BookDto>, which hides pagination metadata (e.g., total pages). Recommendation: Return a Page<BookDto> instead of List<BookDto>:
```suggestion
    public Page<BookDto> getAll(Pageable pageable) {
```